src/index.tsx

```diff
@@ -44,11 +44,15 @@ type AnyPortalNode<C extends Component<any> = Component<any>> = HtmlPortalNode<C
 
 
 const validateElementType = (domElement: Element, elementType: ANY_ELEMENT_TYPE) => {
+    const document = domElement.ownerDocument as any;
+    // Cast document to `any` because Typescript doesn't know about the legacy `Document.parentWindow` field, and also
+    // doesn't believe `Window.HTMLElement`/`Window.SVGElement` can be used in instanceof tests.
+    const window = document.defaultView ?? document.parentWindow; // Fallback on `parentWindow` in order to support IE8 and earlier
```

Comment:
It would be good to use different names, rather than shadowing the `document` and `window` globals, to avoid confusion later on. Maybe `ownerDocument`/`ownerWindow`, or parent or similar?

Also it looks like `ownerDocument` and `defaultView` can be null too, in some niche circumstances. Not totally clear if these are reachable in sane usage, and there's not much we could do about it, but it would be useful to check for that and throw a clear error (something like `Can't validate element, ownerDocument not available` is fine) just in case it is possible. That way this is easier to find & debug, rather than being a confusing 'wrong element type' validation message.

Reply:
Good point about shadowing the real globals - I'll rename them.

Also, good catch about the potential null-ness. I reckon in the case they're null, we could just fall back on the global document/window - what do you think? That would revert to the current behaviour, of always testing instanceof against the global window.HTMLElement and window.SVGElement.

Reply:
Sure, that makes sense! Thanks :+1: